### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by pedzed** [tres-framework/Trestle#20]
Trestle ought to have a documentation. The README isn't sufficient enough. It doesn't have to be split into multiple files per se, but it must be very understandable.

Key points to remember when writing the documentation:
- Examples must be shown. Preferably with its SQL syntax. While it would be (perhaps) insane, it's possible to show every SQL equivalent (MySQL, MS SQL, etc.).
- Even though it's clear for who this product is for, we should provide step-by-step instructions and by as detailed as possible so there are no misconceptions. _Note: detailed does not necessarily mean that it has to be too much text._
- Known bugs must be documented and if possible, an alternative should be provided.

Once the documentation is in a good state, it should be reviewed by ourselves and everything should be followed manually to see if anything is inaccurate, missing or unclear.

We can deliver the best product available, but if the documentation isn't clear, nobody is going to use it: it's hard to figure out how to use the code.

Documentation is not only important for the developers using Trestle, but also who help to make Trestle a better product. In its current state, everything _could_ be remembered, but while the project is growing, it becomes harder and harder. 

Remember: with no (good) documentation, Trestle will hardly get any users.

I'm ready to put time into doing most of the documentation.

**Comment by julianrichen** [comment]
I agree, the README documentation is just a placeholder untill things clean-up.

Should we use the [Tres Docs](https://github.com/tres-framework/docs) or create a [Trestle Wiki](https://github.com/tres-framework/Trestle/wiki)?

I was also considering making a quick video guide/introduction, nothing fancy just "This is Trestle, it's in version 0.x and this is what you can do at the momemt". Print media is apparently dead :p

Docs should be finished before Version 1.0.

**Comment by pedzed** [comment]
> Should we use the Tres Docs or create a Trestle Wiki?

I'm not sure yet. The idea of Tres Docs is to have a central place to get documentation from. This would, in my point of view, make it easier to embed all docs to a website like [tresframework.com](http://tresframework.com). An addition to that is that it's also easier to make all docs consistent with each other.
